lib/pf/constants/role.pm

```diff
@@ -0,0 +1,78 @@
+package pf::constants::role;
+
+=head1 NAME
+
+pf::constants::role - constants for roles
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::constants::role
+
+=cut
+
+use strict;
+use warnings;
+use Readonly;
+
+use Exporter qw(import);
+
+our @EXPORT_OK = qw(
+    @ROLES
+    $REGISTRATION_ROLE
+    $ISOLATION_ROLE
+    $MAC_DETECTION_ROLE
+    $INLINE_ROLE
+    $VOICE_ROLE
+);
+
+=head2 ROLES
+
+Required roles for every switch. Those are reserved words for any additional custom role.
+
+=cut
+
+Readonly::Scalar our $VOICE_ROLE => 'voice';
+Readonly::Scalar our $REGISTRATION_ROLE => 'registration';
+Readonly::Scalar our $ISOLATION_ROLE => 'isolation';
+Readonly::Scalar our $MAC_DETECTION_ROLE => 'macDetection';
+Readonly::Scalar our $INLINE_ROLE => 'inline';
+
+Readonly::Array our @ROLES => (
```

Comment:
I would say via code is enough.

If somebody goes pro style and deletes the voice role directly in the DB, he probably knows what he is doing or deserves to have issues.

Preventing the deletion in the DB may lead to other side effects which we may not want.

Both are good though, as long as pressing the small red button doesn't delete it, I'm happy